scripts/gdal/gdal_bands.py

```diff
@@ -36,16 +59,27 @@ def get_gdal_band_offset(file: str, info: Optional[GdalInfo] = None) -> List[str
 
     bands = info["bands"]
 
-    alpha_band = find_band(bands, "Alpha")
-    alpha_band_info: List[str] = []
-    if alpha_band:
-        alpha_band_info.extend(["-b", str(alpha_band["band"])])
+    band_alpha_arg: List[str] = []
+    if band_alpha := find_band(bands, "Alpha"):
+        band_alpha_arg = ["-b", str(band_alpha["band"])]
 
     # Grey scale imagery, set R,G and B to just the grey_band
-    grey_band = find_band(bands, "Gray")
-    if grey_band:
-        grey_band_index = str(grey_band["band"])
-        return ["-b", grey_band_index, "-b", grey_band_index, "-b", grey_band_index] + alpha_band_info
+    if band_grey := find_band(bands, "Gray"):
+        band_grey_index = str(band_grey["band"])
+        return ["-b", band_grey_index, "-b", band_grey_index, "-b", band_grey_index] + band_alpha_arg
+
+    if band_palette := find_band(bands, "Palette"):
+        palette_channels = len(band_palette["colorTable"]["entries"][0])
+        if palette_channels == 4:
+            return ["-expand", "rgba"]
+        if palette_channels == 3:
+            return ["-expand", "rgb"]
+        get_log().error(
+            "unknown_palette_band_type",
+            palette_channels=palette_channels,
+            first_entry=band_palette["colorTable"]["entries"][0],
+        )
+        raise UnknownBandsException
```

Comment:
I think this will make the entire process exit in error. Is that what we want?

Reply:
I *think* that is what we want. If a band has a colour palette, but the entries in that palette are not either RGB or RGBA values, the `-expand` likely wouldn't work or wouldn't result in a meaningful output.

I haven't tested that as I haven't worked out how to create a paletted TIFF, I've just been using the files from SN9911, whose palette has RGBA values as you can see below in the gdalinfo output.

There is some non-definitive suggestion that at least within GDAL, a colour palette will always be RGBA values, with the following in the documentation for the VRT driver:

> **ColorTable:** This element is parent to a set of Entry elements defining the entries in a color table. Currently only RGBA color tables are supported with c1 being red, c2 being green, c3 being blue and c4 being alpha. The entries are ordered and will be assumed to start from color table entry 0.

If a file has a palette whose colour table *doesn't* have RGB or RGBA values, it feels like it's better just to error out. Effectively "hey this file is too weird, can't process it".

Reply:
Agree with you to stop the process for the specific file that fails. My question was more about letting the other files (that could potentially be standardized with success) rather than preventing them to be processed.

Reply:
Aha, yes! I had just been testing with a single file and looking in at just the gdal_bands stuff, I hadn't considered that the process as a whole was doing more than one file.

I've pushed a new commit which catches the exception, logs that file as failed, and continues processing others. Not sure if it's the best way to structure it overall, I tried to keep it quite minimal and followed the pattern in [create_polygons.py](scripts/create_polygons.py).